packages/node-sdk/runtime/src/api/http-server.ts

```diff
@@ -155,20 +165,65 @@ export const startHttpServer = function* (
     };
   });
 
-  server.get("/addresses", async () => {
-    const result = await runPreparedQuery(
-      getAllAddresses.run(undefined, dbConn),
+  server.get("/addresses", {
+    schema: {
+      tags: ["status"],
+      querystring: PaginationQuerySchema,
+      response: {
+        200: createPaginatedResponseSchema(Type.Object({
+          account_id: Type.Union([Type.Number(), Type.Null()]),
+          address: Type.String(),
+          primary_address: Type.Union([Type.String(), Type.Null()]),
+        })),
+      },
+    },
+  }, async (request) => {
+    const { limit, skip, count } = getPaginationParams(request);
+    const query = request.query as any;
+
+    // Only pass pagination params if they were provided in the request
+    const paginationParams =
+      (query.limit !== undefined || query.skip !== undefined)
+        ? { limit, skip }
+        : {};
+
+    // Only run count query if explicitly requested
+    const addressesPromise = runPreparedQuery(
+      getAllAddresses.run(paginationParams, dbConn),
       "addresses",
     );
-    return result;
+
+    const countPromise = count
```

Comment:
counts are calculated only if count=true param is passed and processed after the main query